src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/EncryptDecrypt.cs

```diff
@@ -324,9 +336,12 @@ private void RsaCryptRoundtrip(RSAEncryptionPadding paddingMode, bool expectSucc
             {
                 if (!expectSuccess)
                 {
-                    Assert.ThrowsAny<CryptographicException>(
+                    Exception ex = Assert.ThrowsAny<Exception>(
                         () => Encrypt(rsa, TestData.HelloBytes, paddingMode));
 
+                    Assert.True(ex is CryptographicException or PlatformNotSupportedException,
+                        "exception is CryptographicException or PlatformNotSupportedException");
```

Comment:
This makes me wonder if we should be doing a CryptographicException wrapping a PNSE, or a CryptographicException, or a PNSE.

They're kind of all wrong.  What did we do for MD5 on wasm?

Reply:
> What did we do for MD5 on wasm?

CryptographicException.

https://github.com/dotnet/runtime/blob/dd6d9294b60ee88c2d64763febcedfed85e0e3d0/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/HashProviderDispenser.Browser.cs#L18

However, for "new" primitives like ChaCha20, we throw PNSE.

https://github.com/dotnet/runtime/blob/dd6d9294b60ee88c2d64763febcedfed85e0e3d0/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/ChaCha20Poly1305.cs#L108-L110

I don't know that MD5 on wasm really feels like it set a pattern I want to follow. I generally don't love the idea "let's throw something other than a platform not supported exception when your platform is not supported".

On the other hand, I don't want to mix and match - it should either be all CE or all PNSE. I learned toward PNSE since that is how we have been lighting up "new" things and your platform just isn't ready for it.